cvprac/cvp_api.py

```diff
@@ -275,7 +275,12 @@ def get_device_by_name(self, fqdn):
                              'queryparam=%s&startIndex=0&endIndex=0' % fqdn,
                              timeout=self.request_timeout)
         if len(data['netElementList']) > 0:
-            device = data['netElementList'][0]
+            for netElement in data['netElementList']:
+                if netElement['fqdn'] == fqdn:
```

Comment:
If the call returns several devices that include the provided fqdn but none of them are a perfect match then device will not be set to an empty dict.

Reply:
Ignore this comment. I mistakenly assumed for/else clause behaved in the opposite manner that it actually does.